styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -518,19 +518,19 @@ private synchronized void tryPollPods() {
 
     final RunState runState = stateManager.get(workflowInstance);
     if (runState == null) {
-      LOG.warn("Pod event for unknown or inactive workflow instance {}", workflowInstance);
+      LOG.debug("Pod event for unknown or inactive workflow instance {}", workflowInstance);
       return Optional.empty();
     }
 
     final Optional<String> executionIdOpt = runState.data().executionId();
     if (!executionIdOpt.isPresent()) {
-      LOG.warn("Pod event for state with no current executionId: {}", podName);
+      LOG.debug("Pod event for state with no current executionId: {}", podName);
       return Optional.empty();
     }
 
     final String executionId = executionIdOpt.get();
     if (!podName.equals(executionId)) {
-      LOG.warn("Pod event not matching current exec id, current:{} != pod:{}",
+      LOG.debug("Pod event not matching current exec id, current:{} != pod:{}",
```

Comment:
How often do we see this one? Timeout is 10 minutes and cleanup delay is 2 minutes. Should be a very rare thing.

Reply:
This could happen also if `retryAfter` is quick enough to start a new retry.